src/client/api/events/node_event.c

```diff
@@ -91,16 +92,16 @@ event_client_handle_t event_init(event_client_config_t *config) {
     return NULL;
   }
   // Prepare client config for mqt network layer
-  mqtt_client_config_t *event_conf = set_client_config(config);
-  if (event_conf == NULL) {
+  client->event_conf = set_client_config(config);
```

Comment:
it can simply assign the pointer directly, `set_client_config` is redundant, isn't it?
```suggestion
  client->event_conf = config;
```
therefore, freeing `client->event_conf` is no needed in `event_destory()`.

Reply:
Yes, it is reduntant. Removed set_client_config.